endaq/cloud/core.py

```diff
@@ -124,6 +125,64 @@ def get_file(self,
 
         return Dataset(f)
 
+    def download_all_ide_files(self,
+                               output_directory: str = "",
+                               should_download_file_fn: Optional[Callable] = None,
+                               force_reload_file_table: bool = False,
+                               file_limit: int = 100) -> np.ndarray:
+        """
+        Downloads all IDE files from the enDAQ Cloud (up to a specified file limit).
+
+        :param output_directory: The directory to download the ide files to
+        :param should_download_file_fn: A function which accepts a row of the IDE file table and returns
+         a boolean value which indicates if the IDE file should be downloaded or not.  If this function is
+         not given, the default function will always return True.
+        :param force_reload_file_table: If the file table to use as reference for what files exist in the cloud
+         should be recomputed even if it is already stored (as `self.file_table`)
+        :param file_limit: The maximum number of files to download.  If the `force_recompute_file_table` parameter
+         is True then this will also be used to limit the number of files put in the file table it creates.
+        :return: An array of the filenames which were just downloaded
+
+        TO-DO:
+         - Would be nice to have a parameter to get only ones with a certain tag
+         - Maybe Have a blacklist and/or whitelist parameter
+        """
+        if not isinstance(output_directory, str):
+            raise TypeError('the "output_directory" parameter must be given a string,'
+                            f'but was given a value of type{type(output_directory)}')
```

Comment:
Should we generate an error if the output directory is a `Path` object?

Reply:
This has now been changed